### ISSUE TITLE
using intermediates_dir deletes sourced file

### ISSUE THREAD

**Issue by bellma-lilly** [rstudio/rmarkdown#1248]
If I source a file in markdown and use the intermediates_dir option when rendering, the sourced file gets deleted.

Simple example.Rmd: 

````
```{r example}
source("blank.R") 
```
````

then:

```r
rmarkdown::render("example.Rmd",intermediates_dir=".")
```

blank.R is deleted. If I do not use intermediates_dir, it does not get deleted.

I am using R-3.4.0 and I installed rmarkdown from devtools before submitting this issue.

**Comment by yihui** [comment]
I know my reply will disappoint you, but the three arguments of `render()`, `output_file`, `output_dir`, and `intermediates_dir`, seem to have caused a lot of trouble, e.g., https://github.com/rstudio/rmarkdown/issues?utf8=%E2%9C%93&q=is%3Aissue+intermediates_dir+

Unfortunately I don't have time to investigate these issues. I wonder if you have a strong reason not to use the default `intermediates_dir`; if not, I recommend that you stick with the default. That said, I acknowledge that this is likely to be a very bad bug in rmarkdown.

**Comment by bellma-lilly** [comment]
I am able to get by without intermediates_dir.  I thought this would be useful to document for other people, even if it was not fixed.  I didn't see it reported anywhere else and it took me quite a while to convince myself that I wasn't doing something dumb.

**Comment by cderv** [comment]
This is happening only when intermediate directory is the original input directory. I tried to make a reprex:
````r
# create temporary directory to work in for example
dir_temp <- fs::path_temp("rmd")
fs::dir_create(dir_temp)

# create test rmd file
test_rmd <- fs::file_temp("test", dir_temp, ext = ".Rmd")
test_r <- fs::path(dir_temp, "blank.R")
writeLines("#blankfile", test_r)

writeLines('---
title: "R Notebook"
output:
  html_document:
      self_contained: true
---

```{r example}
source("blank.R") 
```

', test_rmd)

fs::dir_ls(dir_temp)

# with default dir it is ok
rmarkdown::render(test_rmd, "html_document")
fs::dir_ls(dir_temp)

# with a custom intermediate dir also
interm_dir <- fs::path(dir_temp, "intdir"); fs::dir_create(interm_dir)
rmarkdown::render(test_rmd, "html_document", intermediates_dir = interm_dir)
fs::dir_ls(dir_temp)

# with intermediate dir as the same as the input dir, it is not ok
rmarkdown::render(test_rmd, "html_document", intermediates_dir = dir_temp)
fs::dir_ls(dir_temp)

# clean
unlink(dir_temp, recursive = TRUE)
```

I think this is because `blank.R` is detected as intermediates in 
```r
intermediates <- c(intermediates, output_format$intermediates_generator(original_input, 
  encoding, intermediates_dir))
```
then it should be copied but, as the intermediate dir is the same as the one where `blank.R` is, it is not.
and `intermediates` are cleaned at the end - the original is deleted. 

I think this is related to an old issue #580 and an old fix in #582, where the approach should be extended maybe.
